core/src/test/java/org/jboss/galleon/test/PmTestBase.java

```diff
@@ -155,6 +162,75 @@ public void main() throws Throwable {
             }
         }
         expectedHomeDir.assertState(installHome);
+
+        assertLog(expectedDiff());
+    }
+
+    private void assertLog(List<String> expected) {
+        if (expected == null) {
+            return;
+        }
+
+        List<String> diff = filterFsDiffOutput();
+
+        assertThat(diff).containsExactlyInAnyOrderElementsOf(expected);
+    }
+
+    private List<String> filterFsDiffOutput() {
+        boolean diffStarted = false;
+        List<String> diff = new ArrayList<>();
+        for (String s : output) {
+            if (s.equals(FsDiff.REPLAYING_CHANGES)) {
+                diffStarted = true;
+                continue;
+            }
+            if (!diffStarted) {
+                continue;
+            }
+            if (!s.startsWith(" ") && !s.startsWith("!")) {
+                // end of diff
+                break;
+            }
+            diff.add(s);
+        }
+        return diff;
+    }
+
+    @Override
+    protected MessageWriter getMessageWriter() {
+        return new DefaultMessageWriter() {
+            @Override
+            public void verbose(Throwable cause, CharSequence message) {
+                output.add(message.toString());
+                super.verbose(cause, message);
+            }
+
+            @Override
+            public void print(Throwable cause, CharSequence message) {
+                output.add(message.toString());
+                super.print(cause, message);
+            }
+
+            @Override
+            public void error(Throwable cause, CharSequence message) {
+                output.add(message.toString());
+                super.error(cause, message);
+            }
+
+            @Override
+            public boolean isVerboseEnabled() {
+                return super.isVerboseEnabled();
+            }
+
+            @Override
+            public void close() throws Exception {
+
+            }
+        };
+    }
+
+    protected List<String> expectedDiff() {
+        return null;
```

Comment:
Could this be moved to UserChangesTestBase, assuming no other test needs this?

Reply:
I moved the changes to UserChangesTestBase now